temporalio/worker/_workflow_instance.py

```diff
@@ -739,7 +740,14 @@ def workflow_memo_value(
         )[0]
 
     def workflow_patch(self, id: str, *, deprecated: bool) -> bool:
+        # We use a previous memoized result of this if present. If this is being
+        # deprecated, we can still use memoized result and skip the command.
+        use_patch = self._patches_memoized.get(id)
+        if use_patch is not None:
+            return use_patch
+
         use_patch = not self._is_replaying or id in self._patches_notified
+        self._patches_memoized[id] = use_patch
         # Only add patch command if never sent before for this ID
         if use_patch and not id in self._patches_sent:
```

Comment:
Don't need patches sent anymore

Reply:
Not sure what you mean, once memoized, we won't enter this path anymore so we don't need to track the extra state.

Reply:
True, fixing

Reply:
Fixed